src/settings/AlertWordScreen.js

```diff
@@ -0,0 +1,123 @@
+/* @flow strict-local */
+
+import React, { PureComponent } from 'react';
+import { View } from 'react-native';
+
+import type { Auth, AlertWordsState } from '../types';
+import { connect } from '../react-redux';
+import { Screen, Input } from '../common';
+import ZulipButton from '../common/ZulipButton';
+import { getAuth } from '../selectors';
+import addAlertWords from '../api/alert_words/addAlertWords';
+import AlertWordList from './AlertWordList';
+import removeAlertWords from '../api/alert_words/removeAlertWords';
+import SearchEmptyState from '../common/SearchEmptyState';
+import ErrorMsg from '../common/ErrorMsg';
+import { showToast } from '../utils/info';
+import * as logging from '../utils/logging';
+
+const styles = {
+  containerStyle: {
+    marginHorizontal: 7,
+    marginTop: 20,
+  },
+  addButtonStyle: {
+    marginTop: 10,
+  },
+};
+
+type State = {|
+  text: string,
+  progress: boolean,
+  errorMessage: string,
+|};
+
+type Props = $ReadOnly<{|
+  auth: Auth,
+  alertWords: AlertWordsState,
+|}>;
+
+/**
+ * Alert words are strings that will be highlighted in messageList
+ */
+class AlertWordScreen extends PureComponent<Props, State> {
+  state = {
+    text: '',
+    progress: false,
+    errorMessage: '',
+  };
+
+  handleTextChange = (text: string) => {
+    this.setState({
+      text,
+    });
+  };
+
+  handleSubmit = async () => {
+    const { text, errorMessage } = this.state;
+    const { alertWords, auth } = this.props;
+
+    if (alertWords.includes(text.trim())) {
+      this.setState({ errorMessage: 'This word is already present' });
+    } else {
+      this.setState({ progress: true });
+      try {
+        await addAlertWords(auth, [text]);
+        this.setState({ text: '' });
+        showToast('Created a new alert word');
```

Comment:
This toast only serves to give the user [alarm fatigue](https://en.wikipedia.org/wiki/Alarm_fatigue). The alert word appears in the list; there shouldn't be a need for any further notifications.